client/packages/invoices/src/Prescriptions/DetailView/History/HistoryModal.tsx

```diff
@@ -0,0 +1,121 @@
+  InvoiceNodeStatus,
+  useIntlUtils,
+  InvoiceLineNode,
+} from '@openmsupply-client/common';
+import { Draft } from '../../..';
+
+import { useHistoryColumns } from './columns';
+import { usePrescriptionList } from '../../api';
+
+interface HistoryModalModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  draft: Draft | null;
+  mode: ModalMode | null;
+  patientId: string;
+}
+
+export interface HistoryItem {
+  id: string;
+  itemName: string;
+  unitQuantity: number;
+  directions: string;
+  date: Date | null;
+  prescriber: string;
+}
+
+export const HistoryModal: React.FC<HistoryModalModalProps> = ({
+  isOpen,
+  onClose,
+  patientId,
+}) => {
+  const {
+    query: { data, isLoading },
+  } = usePrescriptionList({
+    filterBy: {
+      otherPartyId: { equalTo: patientId },
+      status: { notEqualTo: InvoiceNodeStatus.New },
+    },
+    sortBy: { key: 'pickedDatetime', direction: 'desc' },
+    first: 20,
+  });
+  const columns = useHistoryColumns();
+  const t = useTranslation();
+  const { getLocalisedFullName } = useIntlUtils();
+  const { Modal } = useDialog({ isOpen, onClose, disableBackdrop: true });
+
+  const historyData = useMemo(() => {
+    const rawData = data?.nodes;
+    if (!rawData) return [];
+
+    const compiled: HistoryItem[] = [];
+
+    rawData.forEach(prescription => {
+      const { clinician, pickedDatetime, lines } = prescription;
+      compiled.push(
+        ...combineCommonLines(lines.nodes as InvoiceLineNode[]).map(line => ({
+          id: line.id,
+          itemName: line.itemName,
+          unitQuantity: line.unitQuantity,
+          directions: line.note ?? '',
+          date: pickedDatetime ? new Date(pickedDatetime) : null,
+          prescriber: getLocalisedFullName(
+            clinician?.firstName,
+            clinician?.lastName
+          ),
+        }))
+      );
+    });
+
+    return compiled;
+  }, [data]);
+
+  return (
+    <TableProvider
+      createStore={createTableStore}
+      queryParamsStore={createQueryParamsStore({
+        initialSortBy: { key: 'expiryDate' },
+      })}
+    >
+      <Modal title={t('heading.history')} width={1125} height={600}>
+        <DataTable
+          id="prescription-line-edit"
+          columns={columns}
+          data={historyData}
+          isLoading={isLoading}
+          dense
+        />
+      </Modal>
+    </TableProvider>
+  );
+};
+
+// From a group of lines from a prescription, combines ones for the same item
+// (from different batches), adding the amounts.
+const combineCommonLines = (lines: InvoiceLineNode[]) => {
+  const uniqueItems: Record<
+    string,
+    InvoiceLineNode & { unitQuantity: number }
+  > = {};
+  lines.forEach(line => {
+    const item = uniqueItems?.[line.item.id];
+    if (item) {
+      item.unitQuantity += line.numberOfPacks * line.packSize;
+    } else {
+      uniqueItems[line.item.id] = {
+        ...line,
+        unitQuantity: line.numberOfPacks * line.packSize,
+      };
+    }
+  });
```

Comment:
```suggestion
  lines.forEach(line => {
    let item = uniqueItems?.[line.item.id];
    if (!item) {
          // new item found, default to 0
         item = {
                      ...line,
                      unitQuantity: 0,
                      };
          uniqueItems[line.item.id] = item;
      } 
    
      item.unitQuantity += line.numberOfPacks * line.packSize;    
    }
  });
```

That's probably not quite the right code (haven't tested it), but I prefer this kind of pattern, as the logic for the calculation is in only one place.

In this case I think it's unlikely to change and looks correct to me, so I'm totally fine with what you had, just thought I'd suggest it to see what your thoughts are.

Reply:
Yup, I hear what you're saying. The main difficulty I had here was fighting with Typescript to assign a value to an object indexed with a non-literal propery name (hence the separate `item` intermediate variable).

 But I've taken the suggestion and put the unitQuantity calculation first, so ended up with this: 97c51e191b

Reply:
Nice looks great